### ISSUE TITLE
[wg/audio] Audio Group Charter

### ISSUE THREAD

**Issue by svgeesus** [w3c/strategy#470]
New charter proposal, reviewers please take note.

# Charter Review

[Charter:](https://w3c.github.io/charter-drafts/2024/audio-wg.html)

[diff from previous charter](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fwww.w3.org%2F2022%2F02%2Faudio-2022.html&doc2=https%3A%2F%2Fw3c.github.io%2Fcharter-drafts%2F2024%2Faudio-wg.html)
[diff from template](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fw3c.github.io%2Fcharter-drafts%2Fcharter-template.html&doc2=https%3A%2F%2Fw3c.github.io%2Fcharter-drafts%2F2024%2Faudio-wg.html)

What kind of charter is this? Check the relevant box / remove irrelevant branches. 

* Existing 
 - [x] Existing WG recharter 

Horizontal Reviews: apply the Github label "Horizontal review requested" to request reviews for accessibility (a11y), internationalization (i18n), privacy, security, and TAG. **Also add a "card" for this issue to the [Strategy Funnel](https://github.com/w3c/strategy/projects/2).**

Communities suggested for outreach: [Audio CG]()

Known or potential areas of concern: _none_

Where would charter proponents like to see issues raised? [GitHub](https://github.com/w3c/charter-drafts)

Anything else we should think about as we review? 

Note: proposed chairs should be copied @... on this issue.

@hoch @mdjp

**Comment by himorin** [comment]
- last entry in history table lack a link to charter (current running one) and extension
- top leading text has typo `isis` in `The mission of the Audio Working Group isis to add`

**Comment by svgeesus** [comment]
@himorin thanks, [fixed](https://github.com/w3c/charter-drafts/commit/b2df2a60cb6c265d6e00143d7667153d3fb59a03)

**Comment by svgeesus** [comment]
@simoneonofri any comments from a security perspective?

**Comment by simoneonofri** [comment]
I have read the security consideration sections, which seem well-reasoned.

A further security consideration is that the element could be used to make the browser process a file with a format that contains an attack on the libraries to then encode/decode the file.

For instance:
 - Vulnerability on [VLC](https://www.cvedetails.com/vulnerability-list/vendor_id-5842/product_id-9978/Videolan-Vlc-Media-Player.html) linked to the reproduction of media files.
 - Crash problem on a browser related to reproducing [a very large file](https://github.com/katspaugh/wavesurfer.js/issues/1001)
Or even potential [cross-origin-read issues ](https://issues.chromium.org/issues/40095913)

**Comment by hoch** [comment]
@simoneonofri Thanks for your feedback!

However, vulnerabilities on VLC and potential cross-origin-read issues are related to the [AudioElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement#specifications), which is not under the Audio WG's purview. See https://www.w3.org/2022/02/audio-2022.html.

Regarding the OOM crash problem, do you believe it is a security issue? In terms of the functionality of API, the WG believes the WebCodec is the right solution for such use case. See https://github.com/WebAudio/web-audio-api/issues/2321#issuecomment-829399855.

**Comment by simoneonofri** [comment]
> @simoneonofri Thanks for your feedback!

You're welcome @hoch

> However, vulnerabilities on VLC and potential cross-origin-read issues are related to the [AudioElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement#specifications), which is not under the Audio WG's purview. See https://www.w3.org/2022/02/audio-2022.html.

Thank you for clarifying the scope. I understood it as an 'input' element where potentially malicious date streams could be received.
 
> Regarding the OOM crash problem, do you believe it is a security issue? In terms of the functionality of API, the WG believes the WebCodec is the right solution for such use case. See [WebAudio/web-audio-api#2321 (comment)](https://github.com/WebAudio/web-audio-api/issues/2321#issuecomment-829399855).

It depends on the reason for the crash. If it is for a memory allocation problem, it is possible.

About the specific issue, the expected behavior can be to "fail securely," often, when there is a malformed input, discarding it is an option; alternatively, it can be sanitized, but this may be complex/risky.

**Comment by hoch** [comment]
> About the specific issue, the expected behavior can be to "fail securely," often, when there is a malformed input, discarding it is an option; alternatively, it can be sanitized, but this may be complex/risky.

First, to clarify the scope of the question I assume you're referring to `decodeAudioData()` method.

In Chromium, it is safe to crash on OOM and IMO it does not pose a security problem. Do we believe this is somehow documented in the spec? I thought this is more of an implementation issue.

**Comment by romankulkovsf** [comment]
I need help

**Comment by svgeesus** [comment]
@simoneonofri if you are now satisfied, please mark the security review as completed

@plh any additional comment from PING?

**Comment by plehegar** [comment]
No objection from PING to the charter

**Comment by svgeesus** [comment]
[AC Review started](https://lists.w3.org/Archives/Member/w3c-ac-members/2024JulSep/0010.html)

**Comment by himorin** [comment]
considering [change around proposed recommendation](https://github.com/w3c/process/issues/861), first sentence in success criteria as `In order to advance to Proposed Recommendation,` seems to be better reverted into the original charter template?

**Comment by svgeesus** [comment]
The charter template was [only updated last week](https://github.com/w3c/charter-drafts/commit/f974b8373ed74abbb7f313d5028f18e52abcecf1) but yes it would be better to fold in that change.

**Comment by svgeesus** [comment]
The charter has been updated to fold in that change

**Comment by plehegar** [comment]
[Announced](https://lists.w3.org/Archives/Member/w3c-ac-members/2024OctDec/0037.html)
